src/domUpdates.js

```diff
@@ -224,8 +224,26 @@ const domUpdates = {
     pastStays.innerHTML = '';
     upcomingStays.innerHTML = '';
 
+    // no future/present bookings
+    const findUpcomingBookings = bookings.filter(booking =>
+      dayjs(booking.booking.date).isAfter(currentDate)
+    );
+    const findPresentBookings = bookings.filter(booking =>
+      dayjs(booking.booking.date).isSame(currentDate)
+    );
+
+    if ((findUpcomingBookings.length + findPresentBookings.length) === 0) {
+      upcomingStays.innerHTML += `
+        <article class="past-booking-card">
+          <p>You have no upcoming stays.</p>
+        </article>
+      `;
+    }
```

Comment:
Hey, nice work implementing some error handling here! It's a good idea to give the users some sort of message if no information is displaying on a page.

I see that this logic, and subsequent rendering, is within a larger 'renderUserDashboard()' method. Could this be taken out and and encapsulated into a smaller function to increase SRP?

Reply:
Yes definitely! This function is huuuuge at the moment and has a lot of logic in it. It's on my refactoring list.